src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -20,9 +20,10 @@ public abstract partial class Array : ICloneable, IList, IStructuralComparable,
     {
         // We impose limits on maximum array length in each dimension to allow efficient
         // implementation of advanced range check elimination in future.
-        // Keep in sync with vm\gcscan.cpp and HashHelpers.MaxPrimeArrayLength.
-        // The constants are defined in this method: inline SIZE_T MaxArrayLength(SIZE_T componentSize) from gcscan
+        // Keep in sync with vm\gchelpers.cpp and HashHelpers.MaxPrimeArrayLength.
+        // The constants are defined in this method: inline SIZE_T MaxArrayLength(SIZE_T componentSize) from gchelpers
         // We have different max sizes for arrays with elements of size 1 for backwards compatibility
+        // Wrapped by GetMaxLength<T>()
         internal const int MaxArrayLength = 0X7FEFFFFF;
         internal const int MaxByteArrayLength = 0x7FFFFFC7;
```

Comment:
Can these be made private now?

Reply:
Should. Can also help latter consumers not get confused.